lib/PhasarLLVM/ControlFlow/Resolver/OTFResolver.cpp

```diff
@@ -144,18 +140,74 @@ OTFResolver::resolveFunctionPointer(llvm::ImmutableCallSite CS) {
         if (P->getType()->isPointerTy() &&
             P->getType()->getPointerElementType()->isFunctionTy()) {
           if (const auto *F = llvm::dyn_cast<llvm::Function>(P)) {
-            if (matchesSignature(F, FTy)) {
+            if (matchesSignature(F, FTy, false)) {
               Callees.insert(F);
             }
           }
         }
+        std::vector<const llvm::GlobalVariable *> GlobalVariableWL;
+        std::stack<const llvm::ConstantAggregate *> ConstantAggregateWL;
+        if (auto *CE = llvm::dyn_cast<llvm::ConstantExpr>(P)) {
+          // Unfortunately this allocates
+          auto *AsI = CE->getAsInstruction();
+          for (auto &Op : AsI->operands()) {
+            if (auto *GVOp = llvm::dyn_cast<llvm::GlobalVariable>(Op)) {
+              GlobalVariableWL.push_back(GVOp);
+            }
+          }
+          AsI->deleteValue();
+        }
+        if (auto *GVP = llvm::dyn_cast<llvm::GlobalVariable>(P)) {
+          GlobalVariableWL.push_back(GVP);
+        }
+        for (auto *GV : GlobalVariableWL) {
+          if (!GV->hasInitializer()) {
+            continue;
+          }
+          auto InitConst = GV->getInitializer();
+          if (auto *InitConstAggregate =
+                  llvm::dyn_cast<llvm::ConstantAggregate>(InitConst)) {
+            ConstantAggregateWL.push(InitConstAggregate);
+          }
+        }
+        while (!ConstantAggregateWL.empty()) {
+          auto ConstAggregateItem = ConstantAggregateWL.top();
+          ConstantAggregateWL.pop();
```

Comment:
See next comment about possible infinite loop:  You can avoid it by keeping track of the worklist items and avoid re-processing.  I admit, I don't fully understand what's going on here, so please review closely!  I can say that without this change, I have a sample program which appear to fall into an infinite loop in this section of code.
```suggestion
        std::set<const llvm::ConstantAggregate *> VisitedConstantAggregates;
        while (!ConstantAggregateWL.empty()) {
          auto ConstAggregateItem = ConstantAggregateWL.top();
          ConstantAggregateWL.pop();
          // We may have already processed the item, avoid an infinite loop:
          if (!VisitedConstantAggregates.insert(ConstAggregateItem).second)
            continue;
```

Reply:
I just pushed a fix that is almost identical to your suggestion. Thanks again!

Reply:
I verified the application analysis does not get stuck in an infinite loop now with the latest version.  Thanks!